code/modules/reagents/chemistry/holder/holder.dm

```diff
@@ -546,6 +546,18 @@
 	var/S = specific_heat()
 	chem_temp = clamp(chem_temp * (J / (S * total_volume)), 2.7, 1000)
 
+/// Applies add_reagent modified by the target's bio armor stat
+/datum/reagents/proc/attack_add_reagent(mob/living/attacker, mob/living/carbon/human/target, reagent, amount, list/data=null, reagtemp = 300, no_react = 0, safety = 0, no_overdose = FALSE)
```

Comment:
This is giga cursed, you don't need this proc at all.
Use the modify by armour proc or whatever I called it, it's used for all armour modifications to numbers.